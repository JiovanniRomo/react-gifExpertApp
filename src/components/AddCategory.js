import React, { useState } from 'react'
import PropTypes from 'prop-types';

//Este componente solo nos ayuda a crear nuevas categorias

export const AddCategory = ({setCategory}) => {


    const [inputValue, setinputValue] = useState('');

    const handleInputCange = (e) => {
        setinputValue(e.target.value);
    }


    //Prevenimos el refresh del navegador
    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 2){
            setCategory(cats => [inputValue, ...cats]);
            setinputValue(''); //esto es para no hacer una doble inserci√≥n, borra lo que el usuario ya escribio
        }


    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type='text'
                value={inputValue}
                onChange={ handleInputCange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategory: PropTypes.func.isRequired
}
