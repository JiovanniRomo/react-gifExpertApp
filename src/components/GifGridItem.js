import React from 'react';
import PropTypes from 'prop-types';

//Este componente solo crea las cracteristicas de cada gif

export const GifGridItem = ({title, url}) => {

    return (
        <div className='card animate__animated animate__fadeInRight'>
            <img 
                src={url}
                alt={title}
            />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propType = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
