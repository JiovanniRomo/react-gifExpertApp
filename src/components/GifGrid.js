import React from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

//Este componente recopila todos los items para mostrarlos

export const GifGrid = ({category}) => {

    const {data:images, loading} = useFetchGifs(category);

    
    return (
        //Como ya estamos retornando 2 objetos(h3, div) , usamos fragment
        //! ES IMPORTANTE USAR CLASSNAME PARA DEFINIR LAS CLASES    
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading</p>}
            <div className='card-grid'>
                {
                    images.map(img => (
                        <GifGridItem 
                            key={img.id}
                            {...img}  //esto hace que las propiedades sean independientes
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propType = {
    category: PropTypes.string.isRequired
}