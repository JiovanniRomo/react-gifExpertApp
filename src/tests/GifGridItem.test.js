import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../components/GifGridItem';


describe('Pruebas en GifGridItem', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/imagen.jpg';
    const wrapper = shallow( <GifGridItem title={title} url={url} /> );
    

    test('should display the component correctly', () => {
        
        expect(wrapper).toMatchSnapshot();
    });

    test('should had a p with the title', () => {
        
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);

    });

    test('should had the img = to the url and alt to the props', () => {

        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    });

    test('should had animate__fadeInRight', () => {
        const div = wrapper.find('div');
        const clase = div.prop('className');

        expect(clase.includes('animate__fadeInRigh')).toBe(true);
        // se espera que clase tenga 'animate__fadeInRigh' y sea true, es decir, que s√≠ la incluya

    });

});



