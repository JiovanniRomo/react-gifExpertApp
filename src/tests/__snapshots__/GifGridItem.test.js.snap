// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem should display the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeInRight"
>
  <img
    alt="Un titulo"
    src="https://localhost(imagen.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
